src/tools/singlejar/BUILD

```diff
@@ -232,12 +232,13 @@ cc_test(
         ":test1",
         ":test2",
         "@local_jdk//:jar",
-        "@local_jdk//:jdk-default",
+        "@local_jdk//:jdk",
```

Comment:
Ideally, this would be fixed not to use @local_jdk. Also, did this test not actually run before? There's no :jdk-default target, I think.

Reply:
> Ideally, this would be fixed not to use `@local_jdk`

So what is the alternative?

I also have been wondering whether the singlejar tests were even tested before, and whether the `//src/tools/singlejar/BUILD` in Bazel is different from the Google internal one (because some test files are never included in the `BUILD` file and see this comment: https://github.com/bazelbuild/bazel/pull/6336#discussion_r223815429).

Reply:
I believe `//external:jdk` and `//external:jar` would be more desirable. Alas, I'm not at all familiar with these JDK rules. I'm consulting with @lberki for advice on the internal review.

Reply:
Me! If all the test needs is to run `jar`, do this instead:

```
cc_test(
  name="test",
  data=["@bazel_tols//tools/jdk:current_java_runtime"],
  toolchains=["@bazel_tools//tools/jdk:current_java_runtime"],
  copts=["-DJAR=$(JAVA)/bin/jar"],
)
```

This way, you get a path to whatever target JDK is used (and not to the result of autodetection), which is the principled thing to do.

Reply:
I think we need to add `.exe` at for the `-D` flag. Windows also requires different kind of `"` escaping to pass string constant in `-D` as `const char[]`. I will handle that in the second PR, just do whatever that works for Unix first.

Reply:
Sounds like a plan. You can use `select({"@bazel_tools//platforms:windows": ... })` or something like that to handle that case.

Reply:
@rongjiecomputer : shall I apply @lberki 's suggestion (the cc_test rule) to the code and submit internally like so?

Reply:
Sure.